internal/util/update/update.go

```diff
@@ -314,7 +314,12 @@ func (u Command) updatePackage(subPkgPath, localPath, updatedPath, originPath st
 	}
 
 	originExists := isRootPkg
-	if !isRootPkg {
+	if isRootPkg {
+		_, err := os.Stat(originPath)
```

Comment:
Why is this needed?

Reply:
Without this branch, the function does not check for the existence of the origin path before attempting to update the package later. The error that's surfaced in that case doesn't give any information about the bad path in the `UpstreamLock` unlike the error returned here.

Reply:
Looking at this now, I actually think the original error here is the right one. In fact, there probably isn't such a thing as an invalid directory, it just happens to be a directory that doesn't exist. And if this is the case, the merge should treat this as an empty origin.
I think we can remove the `invalid dir` test case and this change. Then this PR should be good to merge.

Reply:
Removed.